tests/test_main.py

```diff
@@ -33,6 +34,24 @@ def test_new_environment(tmp_path, solver):
     assert list((tmp_path / "env" / "conda-meta").glob("libzlib-*.json"))
 
 
+def test_install_conda(tmp_path):
+    # Use regex to avoid matching with conda plug-ins or other packages
+    # starting with "conda-"
+    conda_json_regex = re.compile(r"conda-[\d]+\.[\d]+\.[\d]-.*\.json$")
+    run_conda(
+        "create",
+        "-p",
+        tmp_path / "env",
+        "-y",
+        "conda",
+        check=True,
+    )
+    assert list(
+        conda_json_regex.search(str(file))
+        for file in (tmp_path / "env" / "conda-meta").glob("conda-*.json")
+    )
```

Comment:
Are you trying to assert whether `conda` was installed? This might create list of [None, None, ...], which is truthy, right? I think we are better off using `PrefixData` for this, or even `conda list -f ...`. Alternatively, change the generator to:

```suggestion
    assert sum(
        1
        for file in (tmp_path / "env" / "conda-meta").glob("conda-*.json")
        if conda_json_regex.search(str(file))
    ) == 1
```

Reply:
You're right. I changed the generator to `any` here to make the intent more transparent.